### PR TITLE
Created test for /Cloud-pkg-devicecontroller-manager-common.go

### PR DESCRIPTION
**Pull request by vivekbisen04**
/kind test  

## What this PR does / why we need it

This PR significantly improves test coverage for the `./cloud/pkg/devicecontroller/manager/common.go` package. It increases the test coverage from **X% to Y%**, ensuring more comprehensive validation of:

- Device management workflows  
- Edge case handling for device registration and updates  
- Improved reliability of the `common.go` logic  

## Which issue(s) this PR fixes

Part of [lfx-mentorship] Enhance KubeEdge testing coverage initiative **#6101**  

## Screenshots  
image

## Command to test  

- **File path:** `./cloud/pkg/devicecontroller/manager/common.go`  
- **Test execution command:**  

  ```sh
  go test -coverprofile=coverage.out ./cloud/pkg/devicecontroller/manager/...
  go tool cover -html=coverage.out -o coverage.html

### CODE REVIEW

File: cloud/pkg/devicecontroller/manager/common_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	t.Run("Unsupported Object Type", func(t *testing.T) {
+		eventsChan := make(chan watch.Event, 1)
+		handler := NewCommonResourceEventHandler(eventsChan)
+
+		unsupportedObj := "not a runtime object"
+		handler.OnAdd(unsupportedObj, false)
+
+		select {
+		case <-eventsChan:
+			t.Errorf("Event should not be sent for unsupported type")
```
Comment (wbc6080) [inline cloud/pkg/devicecontroller/manager/common_test.go:88]:
```suggestion
			t.Error("Event should not be sent for unsupported type")
```

**Review by wbc6080 — COMMENTED** [review]
Overall LGTM
